src/modules/CSADigitizer/CSADigitizerModule.cpp

```diff
@@ -0,0 +1,337 @@
+/**
+ * @file
+ * @brief Implementation of charge sensitive amplifier digitization module
+ * @copyright Copyright (c) 2017-2020 CERN and the Allpix Squared authors.
+ * This software is distributed under the terms of the MIT License, copied verbatim in the file "LICENSE.md".
+ * In applying this license, CERN does not waive the privileges and immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+#include "CSADigitizerModule.hpp"
+
+#include "core/utils/unit.h"
+#include "tools/ROOT.h"
+
+#include <TFile.h>
+#include <TGraph.h>
+#include <TH1D.h>
+#include <TH2D.h>
+#include <TProfile.h>
+
+#include "objects/PixelHit.hpp"
+
+using namespace allpix;
+
+CSADigitizerModule::CSADigitizerModule(Configuration& config,
+                                               Messenger* messenger,
+                                               std::shared_ptr<Detector> detector)
+    : Module(config, std::move(detector)), messenger_(messenger), pixel_message_(nullptr) {
+    // Enable parallelization of this module if multithreading is enabled
+    enable_parallelization();
+
+    // Require PixelCharge message for single detector
+    messenger_->bindSingle(this, &CSADigitizerModule::pixel_message_, MsgFlags::REQUIRED);
+
+    // Seed the random generator with the global seed
+    random_generator_.seed(getRandomSeed());
+
+    // Set defaults for config variables
+    config_.setDefault<bool>("simple_parametrisation", true);
+    // defaults for the "advanced" parametrisation
+    config_.setDefault<double>("rise_time_constant", Units::get(1e-9, "s"));  
```

Comment:
Now we are setting defaults for everything - this is a bit confusing because all defaults will be added to the configuration written to the output files. Which means when I look into this configuration I will see values for every single parameter - and not just the ones relevant for the model I selected.